### PR TITLE
Rack: Collect target location for redirects

### LINKED ISSUE #5: Rack: Collect target location for redirects
It would be useful if `app.web.request.rack` spans could also store the redirect location for redirect responses. :)

### PR DESCRIPTION
**Pull request by franzliedke**
Fixes #5.

### CODE REVIEW

File: lib/mnemosyne/middleware/rack.rb [hunk 1]
```diff
@@ -109,7 +109,10 @@ def scan_response(trace, response)
 
         return unless headers
 
-        trace.meta[:content_type] = headers['Content-Type']
+        trace.meta.merge!({
+          content_type: headers['Content-Type'],
```
Comment (franzliedke) [inline lib/mnemosyne/middleware/rack.rb]: When `Content-Type` header is missing, this item will no longer be attached. I suppose this should be fine?

File: lib/mnemosyne/middleware/rack.rb [hunk 2]
```diff
@@ -109,7 +109,10 @@ def scan_response(trace, response)
...
+          location: headers['Location'],
```
Comment (franzliedke) [inline lib/mnemosyne/middleware/rack.rb]: @jgraichen Will mnemosyne-server just start showing this in a span's meta information? Currently, it does not seem to include all the meta fields added in this middleware...

Comment (jgraichen) [reply lib/mnemosyne/middleware/rack.rb]: No, it wont.

File: spec/mnemosyne/middleware/rack_spec.rb [hunk 3]
```diff
@@ -70,6 +70,23 @@ def consume(response)
       content_type: 'text/plain'
   end
 
+  context 'with a redirect' do
+    let(:response) do
+      [302, {'Location' => 'http://www.mnemosyne.com'}, ['']]
```
[the remark this answers is not in the document]
Comment (franzliedke) [reply spec/mnemosyne/middleware/rack_spec.rb]: Done.

**Pull request merged**